### PR TITLE
Update tower-spawn-ready to std::future

### PR DESCRIPTION
**Pull request by jonhoo**
Will rebase and mark as ready once #330 lands (which this depends on)

### CODE REVIEW

**Comment by jonhoo** [comment]
Bah, forgot to mark this as a draft. Oh well. I wonder why GitHub won't let me mark a PR as draft after the fact...

File: tower-spawn-ready/src/make.rs [hunk 1]
```diff
@@ -11,8 +17,10 @@ pub struct MakeSpawnReady<S, E = DefaultExecutor> {
 }
 
 /// Builds a SpawnReady with the result of an inner Future.
+#[pin_project]
```
Comment (LucioFranco) [inline tower-spawn-ready/src/make.rs]: we should prob just drop the `E` since all the executor traits are in flux anyways.

**Comment by jonhoo** [comment]
Removed the generic executor parameter in the latest commit.

File: tower-spawn-ready/src/service.rs [hunk 2]
```diff
@@ -26,34 +25,20 @@ enum Inner<T> {
...
     }
 }
 
-impl<T, E, Request> Service<Request> for SpawnReady<T, E>
+impl<T, Request> Service<Request> for SpawnReady<T>
 where
     T: Service<Request> + Send,
     T::Error: Into<Error>,
-    E: BackgroundReadyExecutor<T, Request>,
+    DefaultExecutor: BackgroundReadyExecutor<T, Request>,
```
Comment (LucioFranco) [inline tower-spawn-ready/src/service.rs]: don't think you need this?

Comment (jonhoo) [reply tower-spawn-ready/src/service.rs]: You do, otherwise the call to `spawn` doesn't work (or you'd have to `Box::pin` the future).

**Pull request merged**